src/tests/src/solver/infeasible-problem-analysis/test-unfeasible-problem-analyzer.cpp

```diff
@@ -0,0 +1,207 @@
+    analysis.push_back(std::make_unique<AnalysisMock>(true, hasRun2, hasPrinted2));
+    std::unique_ptr<MPSolver> problem(MPSolver::CreateSolver("GLOP"));
+
+    UnfeasiblePbAnalyzer analyzer(std::move(analysis));
+    BOOST_CHECK(!hasRun1);
+    BOOST_CHECK(!hasPrinted1);
+    BOOST_CHECK(!hasRun2);
+    BOOST_CHECK(!hasPrinted2);
+
+    analyzer.run(problem.get());
+    BOOST_CHECK(hasRun1);
+    BOOST_CHECK(!hasPrinted1);
+    BOOST_CHECK(hasRun2);
+    BOOST_CHECK(!hasPrinted2);
+
+    // only failing analysis will print
+    analyzer.printReport();
+    BOOST_CHECK(hasRun1);
+    BOOST_CHECK(!hasPrinted1);
+    BOOST_CHECK(hasRun2);
+    BOOST_CHECK(hasPrinted2);
+}
+
+BOOST_AUTO_TEST_CASE(test_variable_bounds_consistency)
+{
+    std::unique_ptr<MPSolver> problem(MPSolver::CreateSolver("GLOP"));
+    problem->MakeNumVar(-1, 1, "ok-var");
+    problem->MakeNumVar(1, -1, "not-ok-var");
+
+    VariablesBoundsConsistency analysis;
+    analysis.run(problem.get());
+    auto incorrectVars = analysis.incorrectVars();
+    BOOST_CHECK_EQUAL(incorrectVars.size(), 1);
+
+    auto expected = VariableBounds("not-ok-var", 1, -1);
+    BOOST_CHECK(variableEquals(incorrectVars[0], expected));
+}
+
+/*!
+ * Creates a problem with 2 variables linked by 1 constraint:
+ *  - Variable 1 must be greater than 1
+ *  - Variable 2 must be lesser than -1
+ *  - but if feasible is false, constraint enforces that variable 2 is greater than variable 1 --> infeasible
+ */
+std::unique_ptr<MPSolver> createProblem(const std::string& constraintName, bool feasible)
+{
+    ConstraintSlackAnalysis analysis;
+    std::unique_ptr<MPSolver> problem(MPSolver::CreateSolver("GLOP"));
+    const double inf = problem->infinity();
+    auto var1 = problem->MakeNumVar(1, inf, "var1");
+    auto var2 = problem->MakeNumVar(-inf, -1, "var2");
+    auto constraint = problem->MakeRowConstraint(constraintName);
+    constraint->SetBounds(0, inf);
+    if (feasible) {
+        constraint->SetCoefficient(var1, 1);
+        constraint->SetCoefficient(var2, -1);
+    } else {
+        constraint->SetCoefficient(var1, -1);
+        constraint->SetCoefficient(var2, 1);
+    }
+    return problem;
+}
+
+static const std::string validConstraintNames[] =
+{
+    "BC::hourly::hour<36>",
+    "BC::daily::day<67>",
+    "BC::weekly::week<12>",
+    "FictiveLoads::hour<25>",
+    "AreaHydroLevel::hour<8>",
+};
+
+BOOST_DATA_TEST_CASE(test_slack_variables_with_unfeasible_constraint, bdata::make(validConstraintNames), constraintName)
+{
+    std::unique_ptr<MPSolver> unfeasibleProblem = createProblem(constraintName, false);
+    BOOST_CHECK(unfeasibleProblem->Solve() == MPSolver::INFEASIBLE);
+
+    ConstraintSlackAnalysis analysis;
+    analysis.run(unfeasibleProblem.get());
+    BOOST_CHECK(analysis.hasDetectedInfeasibilityCause());
+}
+
+BOOST_AUTO_TEST_CASE(test_slack_variables_with_unfeasible_constraint_name_ignored)
```

Comment:
A name could be : 
slack variable analysis does not detect unfeasibilty on ill-named constraint